executor/aggfuncs/func_cume_dist.go

```diff
@@ -26,7 +27,7 @@ type cumeDist struct {
 type partialResult4CumeDist struct {
 	curIdx   int
 	lastRank int
-	rows     []chunk.Row
+	rows     [][]*types.Datum
```

Comment:
It's better to use a `Chunk` to buffer the underlying data, and still use `[]chunk.Row` to store the pointers.

Reply:
It's hard to copy some columns of a Row to a new Row/Chunk.
Vectorizing window functions is a good opportunity to optimize this, maybe later.